maud/tests/basic_syntax.rs

```diff
@@ -181,6 +181,34 @@ fn hyphens_in_attribute_names() {
     );
 }
 
+#[test]
+fn string_literals_in_attribute_names() {
+    let result = html! { this "@sentence:-is.not"="false" of-course {} };
+    assert_eq!(
+        result.into_string(),
+        r#"<this @sentence:-is.not="false" of-course></this>"#
+    );
+}
+
+#[test]
+fn raw_string_literals_in_attribute_names() {
+    let result = html! { this r#"@sentence:-is.not"#="false" of-course {} };
+    assert_eq!(
+        result.into_string(),
+        r#"<this @sentence:-is.not="false" of-course></this>"#
+    );
+}
+
+#[test]
+fn other_literals_in_attribute_names() {
+    let result =
+        html! { this b"byte_string"="false" 123="123" 2.5 true 'a'="a" b'b'="b" of-course {} };
+    assert_eq!(
+        result.into_string(),
+        r#"<this byte_string="false" 123="123" 2.5 true a="a" b="b" of-course></this>"#
```

Comment:
I think string literals, and _maybe_ integers, are enough.

Floating point numbers overlap with class shorthand which also uses `.`. There's no use case for a single character. And no other part of Maud accepts byte strings.

Reply:
@lambda-fairy, addressing this issue, to keep `ast::name_to_string()` as an infallible api (always returning `String` without errors) I handled all the cases since I didn't see any issue with also handling `ByteStr` and others so they produce the expected internal string. Current implementation should always produce valid html.

The class shorthand shouldn't be broken since classes starting with numbers are disallowed in CSS so all classes before should still be possible. Also the only time a class starting with a number wouldn't be interpreted as a class would be ` 123.123` (space followed by numbers with dot and no spaces), but `123 .123`, `anything.123` and `.classname123.123` would still parse as a class shorthand for  the class `123`.

Reply:
Having the interpretation change just by adding a space is the confusing part.

Reply:
#398 fixes this issue by running the existing `literal()` function in the parser when encountering a literal, reusing existing error behaviour. I've added `Int` as an accepted literal, but can revoke that change if you decide that `Int` shouldn't be allowed.